### PR TITLE
TST: Don't require downloading of data for tests

### PR DESCRIPTION
**Pull request by richafrank**
to avoid a new trading env needing to download data unnecessarily

### CODE REVIEW

**Comment by Peque** [comment]
I have been getting those HTTP errors too, it seems `ichart.finance.yahoo.com` is down. They say they "will be right back..." though.

**Comment by FreddieV4** [comment]
x-ref https://github.com/quantopian/zipline/issues/1776

File: zipline/testing/core.py [hunk 1]
```diff
@@ -872,8 +871,13 @@ class tmp_trading_env(tmp_asset_finder):
     empty_trading_env
     tmp_asset_finder
     """
+    def __init__(self, load=None, *args, **kwargs):
```
Comment (yankees714) [inline zipline/testing/core.py]: Should we update the docstring?

File: zipline/data/loader.py [hunk 2]
```diff
@@ -162,19 +163,22 @@ def load_market_data(trading_day=None, trading_days=None, bm_symbol='^GSPC'):
...
+        environ,
     )
     benchmark_returns = br[br.index.slice_indexer(first_date, last_date)]
     treasury_curves = tc[tc.index.slice_indexer(first_date, last_date)]
     return benchmark_returns, treasury_curves
 
 
-def ensure_benchmark_data(symbol, first_date, last_date, now, trading_day):
+def ensure_benchmark_data(symbol, first_date, last_date, now, trading_day,
+                          environ=None):
```
Comment (yankees714) [inline zipline/data/loader.py]: Is the `environ` arg ever passed?

File: zipline/testing/core.py [hunk 3]
```diff
@@ -695,11 +697,8 @@ def create_data_portal_from_trade_history(asset_finder, trading_calendar,
 
 
 class FakeDataPortal(DataPortal):
-    def __init__(self, env=None, trading_calendar=None,
+    def __init__(self, env, trading_calendar=None,
                  first_trading_day=None):
-        if env is None:
-            env = TradingEnvironment()
```
Comment (yankees714) [inline zipline/testing/core.py]: 👍

File: zipline/testing/core.py [hunk 4]
```diff
@@ -1486,6 +1490,24 @@ def patched_read_csv(filepath_or_buffer, *args, **kwargs):
...
+def copy_market_data(src_market_data_dir, dest_root_dir):
+    symbol = '^GSPC'
+    filenames = (get_benchmark_filename(symbol), INDEX_MAPPING[symbol][1])
+
+    try:
+        os.makedirs(os.path.join(dest_root_dir, 'data'))
+    except OSError as e:
+        # be happy if someone already created the path
+        if e.errno != errno.EEXIST:
+            raise
```
Comment (yankees714) [inline zipline/testing/core.py]: Could this use `zipline.utils.paths.ensure_directory`?

File: zipline/testing/core.py (hunk 1 above)
Comment (richafrank) [inline zipline/testing/core.py]: 👍

File: zipline/testing/core.py (hunk 4 above)
Comment (richafrank) [inline zipline/testing/core.py]: Oh good call.

File: zipline/data/loader.py (hunk 2 above)
Comment (richafrank) [inline zipline/data/loader.py]: Yea, the call site is [here](https://github.com/quantopian/zipline/pull/1793/files#diff-0ec9e6870d90568b4a83e0daa3a317b1R166).

Comment (yankees714) [inline zipline/data/loader.py]: Ah, thanks!

**Pull request merged**